src/interfaces.rs

```diff
@@ -153,6 +161,13 @@ pub trait ClientLike: Clone + Send + Sized {
     self.inner().server_state.read().server_version()
   }
 
+  /// Override the DNS resolution logic for the client.
+  #[cfg(feature = "dns")]
+  #[cfg_attr(docsrs, doc(cfg(feature = "dns")))]
+  async fn set_resolver(&self, resolver: Arc<dyn Resolve>) {
```

Comment:
Could you reflect this function to the `Pool` implementation? It should update all underlying clients 🤔

Reply:
Good catch, yes definitely.